service/pipeline.ml

```diff
@@ -49,8 +49,8 @@ module Docker_of_builder (Builder : Conf.BUILDER) : Ocaml_ci.S.DOCKER_CONTEXT =
 
   let image_hash = Builder.Image.hash
 
-  let pull ~schedule name =
-    Builder.pull ~schedule name
+  let pull name =
+    Builder.pull ~schedule:weekly name
```

Comment:
As in the `pull` case, this feels like unnecessary specialisation.

Reply:
I think this is part of the global config and should be abstract to `lib`. I think it's not the job of `Lint` of `Opam_build` to set that value.

Reply:
I agree with you, but that wasn't what I was suggesting. 

You could lift this argument from the `Buidler` API to the `Lint` / `Opam_build` APIs, allowing those schedules to be different without needing to construct another `DOCKER_CONTEXT` module. This would define the schedules in `service/pipeline.ml` rather than `service/conf.ml` (which I think is slightly better, since it's one fewer difference between `Current_docker.S` and `Ocaml_ci.S.DOCKER_CONTEXT`).

Reply:
I did it that way because passing that value is not something the current implementation does. My change shows that.
I agree that it's an arbitrary limitation, but I argue that it was already like that is the code.
I think this should be discussed in an other PR and when this is necessary and when we know how this value should change.

Reply:
I think this is a limitation introduced by this PR. The current implementation defines the schedules of `lint` and `build` separately in `service/pipeline.ml` (which is where I would expect to see it):
- https://github.com/ocaml-ci/ocaml-ci/blob/master/service/pipeline.ml#L48
- https://github.com/ocaml-ci/ocaml-ci/blob/master/service/pipeline.ml#L69

Reply:
I put the `schedule` argument back and passed it through. I think this is not the right way and a refactor is needed.
I think the `Docker.pull` should be done in `pipeline.ml` (and hidden in `DOCKER_CONTEXT`)